pkg/printers/json/formatter_test.go

```diff
@@ -106,6 +106,27 @@ func TestFormatter_FormatApplyEvent(t *testing.T) {
 				},
 			},
 		},
+		"resource apply skip error": {
+			previewStrategy: common.DryRunNone,
+			event: event.ApplyEvent{
+				Operation:  event.Unchanged,
+				Identifier: createIdentifier("apps", "Deployment", "", "my-dep"),
+				Error:      errors.New("example error"),
+			},
+			expected: []map[string]interface{}{
+				{
+					"eventType": "resourceApplied",
```

Comment:
This "resourceApplied" eventType feels inaccurate, but "resourceFailed" is wrong too. We could add a new "resourceSkipped", but then it's redundant with the operations.

I think we should come back in another PR and clean up the json printer to use the newer strategy/actuation/reconcile pattern we made for the ResourceGroup status interface.

Reply:
We need to be careful about breaking changes like this in the json printer. Unlike the other output formats, we know users are relying on the format here.

Reply:
This may or may not be breaking, because it would already have been `resourceApplied`, it just wouldn't have had an error. So it's only breaking if the caller is looking for the error first and not the "ApplyUnspecified" operation. Granted the likelihood is high, because there's no ApplyFailed... So maybe I should add ApplyFailed in another PR as a prerequisite.